src/EvalGraph.purs

```diff
@@ -196,6 +201,33 @@ eval_progCxt (ProgCxt { primitives, mods, datasets }) =
       v <- eval γ e empty
       pure $ γ <+> maplet x v
 
+evalCmt :: forall m. MonadWithGraphAlloc m => Env Vertex -> DocOpt Expr Vertex -> m (DocOpt Val Vertex)
+evalCmt _ None = pure None
+evalCmt γ (Doc tokens) = Doc <$> sequence (map evalToken tokens)
+   where
+   evalToken :: DocCommentElem Expr Vertex -> m (DocCommentElem Val Vertex)
+   evalToken (Token s) = pure $ Token s
+   evalToken (Unquote e) = Unquote <$> (eval γ e empty)
+
+new'
+   :: forall m
+    . MonadWithGraphAlloc m
+   => Env Vertex
+   -> (Vertex -> DocOpt Val Vertex -> BaseVal Vertex -> Val Vertex)
+   -> Set Vertex
+   -> DocOpt Expr Vertex
+   -> BaseVal Vertex
+   -> m (Val Vertex)
+new' _ constr αs None bv = new (\αs' -> \bv' -> constr αs' None bv') αs bv
+new' γ constr αs doc bv = do
```

Comment:
Maybe `u` is a better convention than `bv` for the base val (I don't think we use `bv` anywhere else)